dipy/reconst/cti.py

```diff
@@ -0,0 +1,634 @@
+                raise ValueError(msg)
+
+        self.args = args
+        self.kwargs = kwargs
+
+        self.min_signal = self.kwargs.pop('min_signal', None)
+        if self.min_signal is None:
+            self.min_signal = MIN_POSITIVE_SIGNAL
+        elif self.min_signal <= 0:
+            msg = "The `min_signal` key-word argument needs to be strictly"
+            msg += " positive."
+            raise ValueError(msg)
+
+        self.design_matrix = design_matrix(self.gtab1, self.gtab2)
+        self.inverse_design_matrix = np.linalg.pinv(self.design_matrix)
+
+        tol = 1e-6
+        self.min_diffusivity = tol / -self.design_matrix.min()
+        # self.convexity_constraint = fit_method in {'CLS', 'CWLS'}
+        # if self.convexity_constraint:
+        #     self.cvxpy_solver = self.kwargs.pop('cvxpy_solver', None)
+        #     self.convexity_level = self.kwargs.pop('convexity_level', 'full')
+        #     msg = "convexity_level must be a positive, even number, or 'full'."
+        #     if isinstance(self.convexity_level, str):
+        #         if self.convexity_level == 'full':
+        #             self.sdp_constraints = load_sdp_constraints('dki')
+        #         else:
+        #             raise ValueError(msg)
+        #     elif self.convexity_level < 0 or self.convexity_level % 2:
+        #         raise ValueError(msg)
+        #     else:
+        #         if self.convexity_level > 4:
+        #             msg = "Maximum convexity_level supported is 4."
+        #             warnings.warn(msg)
+        #             self.convexity_level = 4
+        #         self.sdp_constraints = load_sdp_constraints(
+        #             'dki', self.convexity_level)
+        #     self.sdp = PositiveDefiniteLeastSquares(22, A=self.sdp_constraints)
+
+        self.weights = fit_method in {'WLS', 'WLLS', 'UWLLS', 'CWLS'}
+        self.is_multi_method = fit_method in ['WLS', 'OLS', 'UWLLS', 'ULLS',
+                                              'WLLS', 'OLLS', 'CLS', 'CWLS']
+
+    def fit(self, data, mask=None): #here data is cti_params of shape : (n, 48) 
+        """ Fit method of the CTI model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[-1]
+
+        """
+        data_thres = np.maximum(data, self.min_signal)
+        params = self.fit_method(self.design_matrix, data_thres,
+                                 *self.args, **self.kwargs)
+        #we need to somehow obtian cti_params from data (it's actually cti_pred_signals)
+        print('this is data.shape: ',data.shape)
+        return CorrelationTensorFit(self, params) #ig there's a need to somehow define cti_params here. 
+    
+    @multi_voxel_fit
+    def multi_fit(self, data_thres, mask=None):
```

Comment:
Unlike the case of the DKI, we don't need to have a separate multi_fit function since we should not aim to implement the NLS fit in this first PR. Let's only focus on the OLS and WLS fits.

Reply:
Alternatively, you can leave the multi_fit function here, but make sure that it is being called by the fit function (defined line 206 in analogous that it is done in dki.py)